syncapi/streams/stream_pdu.go

```diff
@@ -276,7 +276,7 @@ func (p *PDUStreamProvider) getJoinResponseForCompleteSync(
 	if !wantFullState {
 		excludingEventIDs = make([]string, 0, len(recentStreamEvents))
 		for _, event := range recentStreamEvents {
-			if event.StateKey() != nil {
+			if event.StateKey() != nil && event.Type() != "m.room.name" {
```

Comment:
I'm a bit confused as to how/why this is a solution. The idea of this block of code (despite the seemingly incomplete comment) is so that if state events already appear in the `"timeline"` section of the room response, we don't then repeat them a second time in the `"state"` section too. 

Have you been able to verify that the `m.room.name` event was indeed missing from the timeline on a complete sync?

Reply:
Yes, I did. it's missing from timeline when the client gets the response. This is because for a recently created room, the recentStreamEvents contains m.room.name and it is excluded here. Then in line 298-299, it is also excluded from recentEvents because the user just joined the room:

			if membership == "join" {
				joinEventIndex = i

So at this point, m.room.name is filtered from both stateEvents and recentEvents. Consequently, an invited user who just joined this room will not see the m.room.name event. The room name is always "Empty room"

Reply:
@neilalexander any thoughts on how to fix this differently?

Reply:
Did you see my other comment on whether #2281 solves your problem?

Reply:
Yes, this works better. Thanks for the fix.